src/plugins/event-plugins/__tests__/ResourcePlugin.test.ts

```diff
@@ -235,4 +235,28 @@ describe('ResourcePlugin tests', () => {
 
         expect(record).not.toHaveBeenCalled();
     });
+
+    test('when entry name is an invalid url then resource is ignored', async () => {
```

Comment:
issue: We want to test that when the entry name is an invalid URL, then a PerformanceResourceTiming event is recorded.

Reply:
Is there a reason why we can't record the event if the URL is invalid?

The only reason we parse the URL is to check to see if the resource is a (PutRumEvents) call to the RUM endpoint, in which case we don't want to record the resource. Without this check, we wouldn't know or care if the URL is invalid.

Reply:
https://github.com/shubhsheth/aws-rum-web/pull/2

Reply:
I kinda get what what Quinn is saying. Please review https://github.com/shubhsheth/aws-rum-web/pull/3

Reply:
Thanks @williazz! I merged in your changes and updated unit tests.